RELEASE-NOTES.md

```diff
@@ -17,10 +17,10 @@ The format of this file is based on [Keep a Changelog](http://keepachangelog.com
 
 The Axom project release numbers follow [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
-## [Unreleased] - Release date yyyy-mm-dd
```

Comment:
@rhornung67 -- I removed the empty `Unreleased` section for the release. We'll add it back when we merge main to develop